src/widget/wlibrarytableview.cpp

```diff
@@ -100,33 +100,71 @@ void WLibraryTableView::moveSelection(int delta) {
             if (row + 1 < pModel->rowCount()) {
                 selectRow(row + 1);
             }
-
             delta--;
         } else {
             // i is negative, so we want to move the highlight up
             int row = current.row();
             if (row - 1 >= 0) {
                 selectRow(row - 1);
             }
-
             delta++;
         }
     }
 }
 
 void WLibraryTableView::saveVScrollBarPos(TrackModel* key){
+    qDebug()<<"";
+    qDebug()<<"    WLibraryTableView::saveVScrollBarPos";
+    qDebug()<<"       m_vScrollBarPosValues["<<key<<"] ="<<verticalScrollBar()->value();
     m_vScrollBarPosValues[key] = verticalScrollBar()->value();
+    if (currentIndex().isValid()) {
+        // Act only when there is a focused cell. For example, there is none
+        // when library features are added initially after start.
+        qDebug()<<"        QModelIndexList selection = selectionModel()->selectedRows();";
+        QModelIndexList selection = selectionModel()->selectedRows();
+        if (selection.size() > 0) {
+            m_selectedIndices[key] = selection;
+        }
+    }
 }
 
 void WLibraryTableView::restoreVScrollBarPos(TrackModel* key){
+    qDebug()<<"";
+    qDebug()<<"    WLibraryTableView::restoreVScrollBarPos";
     updateGeometries();
 
-    if (m_vScrollBarPosValues.contains(key)){
+    if (m_vScrollBarPosValues.contains(key)) {
+        qDebug()<<"   restore previous vertical scrollbar pos from m_vScrollBarPosValues["<<key<<"]";
         verticalScrollBar()->setValue(m_vScrollBarPosValues[key]);
-    }else{
+    } else {
+        qDebug()<<"   store & set default vertical scrollbar position";
         m_vScrollBarPosValues[key] = 0;
         verticalScrollBar()->setValue(0);
     }
+    if (m_selectedIndices.contains(key)) {
+        QModelIndexList selection = m_selectedIndices[key];
+        // See documentation of failed commands in https://github.com/mixxxdj/mixxx/pull/2378
+        // before restoring the previous selection first select the first row
+        // in order to set focus for arrow key navigation
+        if (index.isValid()) {
+            qDebug()<<"   select row "<<selection.first().row();
+            selectRow(selection.first().row());
+        }
+        // before looping over indices, remove first index we just restored
+        if (selection.size() > 1) {
+            selection.removeFirst();
+        } else {
+            // return when there was only one index stored
+            return;
+        }
+        foreach (QModelIndex index, selection) {
+            if (index.isValid()) {
+                qDebug()<<"    index "<<index<<" isValid(), select.";
+                selectionModel()->select(index,
```

Comment:
Why does 
Select(selection) not work?

Reply:
oh,.. I'll try that again.

Reply:
though I can't assess what happens when the tabel changed in the meantime and there is an now invalid index in the selection.

Reply:
there is `isValid()` for `QItemSelectionRange` (continuous selection) but not for `QItemSelection`